### PR TITLE
dx-991 added info on token deletion

### PR DESCRIPTION
**Pull request by marinafrydenb**

### CODE REVIEW

File: _includes/card-purchase.md [hunk 1]
```diff
@@ -296,6 +296,51 @@ Content-Type: application/json
 | └─➔&nbsp;`href`                   | `string`     | The target URI to perform the operation against.                                                                                                                                                                                                                                                                                                           |
 | └─➔&nbsp;`rel`                    | `string`     | The name of the relation the operation has to the current resource.                                                                                                                                                                                                                                                                                        |
 
+
+Customers should be able to set `paymentToken` to deleted if the consumer
```
Comment (payex-dev) [inline _includes/card-purchase.md:300]: Codacy found an issue: [[no-consecutive-blank-lines] Remove 1 line before node](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721&bid=19787468)

**Review by NullableInt — COMMENTED** [review]
Looks good! But could it be made into an include?
Maybe with default `recurrenceToken` and optionally setting it to `paymentToken`?

File: _includes/card-purchase.md [hunk 2]
```diff
@@ -296,6 +296,51 @@ Content-Type: application/json
 | └─➔&nbsp;`href`                   | `string`     | The target URI to perform the operation against.                                                                                                                                                                                                                                                                                                           |
 | └─➔&nbsp;`rel`                    | `string`     | The name of the relation the operation has to the current resource.                                                                                                                                                                                                                                                                                        |
 
+
+Customers should be able to set `paymentToken` to deleted if the consumer 
+deletes their agreements with their customer.
```
Comment (asbjornu) [inline _includes/card-purchase.md:301]: We need to explain how the operation that describes how to delete a token is found. What is its `rel` name? Please provide an example and update the `<OPERATION NAME>` in the description.

```suggestion
Payers should be able to delete payment tokens that are associated to
them. To delete a `paymentToken`, perform the `<OPERATION NAME>`
operation.

```

File: _includes/payment-order-other-features.md [hunk 3]
```diff
@@ -860,6 +860,42 @@ sequenceDiagram
     end
 ```
 
+Customers should be able to set `recurrenceToken` to deleted if the consumer 
+deletes their agreements with their customer.
```
Comment (asbjornu) [inline _includes/payment-order-other-features.md:864]: We need to explain how the operation that describes how to delete a token is found. What is its `rel` name? Please provide an example and update the `<OPERATION NAME>` in the description.

```suggestion
Payers should be able to delete payment tokens that are associated to
them. To delete a `recurrenceToken`, perform the `<OPERATION NAME>`
operation.

```

File: _includes/card-purchase.md [hunk 4]
```diff
@@ -296,6 +296,51 @@ Content-Type: application/json
...
+Customers should be able to set `paymentToken` to deleted if the consumer 
+deletes their agreements with their customer. 
+Note that the value of `state` must be `Deleted` when deleting the token. 
+No other states are supported.
+
+{:.code-header}
+**Request**
+
+```http
+POST /psp/creditcard/payments HTTP/1.1
```
Comment (asbjornu) [inline _includes/card-purchase.md:309]: This URL does not seem correct.

File: _includes/payment-order-other-features.md [hunk 5]
```diff
@@ -860,6 +860,42 @@ sequenceDiagram
...
+Customers should be able to set `recurrenceToken` to deleted if the consumer 
+deletes their agreements with their customer. 
+Note that the value of `state` must be `Deleted` when deleting the token. 
+No other states are supported.
+
+{:.code-header}
+**Request**
+
+```http
+POST /psp/paymentorders/payments HTTP/1.1
```
Comment (asbjornu) [inline _includes/payment-order-other-features.md:872]: This URL does not seem correct.

File: _includes/recur.md [hunk 6]
```diff
@@ -37,3 +37,46 @@ Content-Type: application/json
     }
 }
 ```
+
+Customers should be able to set `recurrenceToken` to deleted if the consumer 
+deletes their agreements with their customer.
```
Comment (asbjornu) [inline _includes/recur.md:42]: We need to explain how the operation that describes how to delete a token is found. What is its `rel` name? Please provide an example and update the `<OPERATION NAME>` in the description.

```suggestion
Payers should be able to delete payment tokens that are associated to
them. To delete a `recurrenceToken`, perform the `<OPERATION NAME>`
operation.

```

File: payments/invoice/other-features.md [hunk 7]
```diff
@@ -177,6 +177,46 @@ Content-Type: application/json
 }
 ```
 
+Customers should be able to set `recurrenceToken` to deleted if the consumer 
+deletes their agreements with their customer.
```
Comment (asbjornu) [inline payments/invoice/other-features.md:181]: We need to explain how the operation that describes how to delete a token is found. What is its `rel` name? Please provide an example and update the `<OPERATION NAME>` in the description.

```suggestion
Payers should be able to delete payment tokens that are associated to
them. To delete a `recurrenceToken`, perform the `<OPERATION NAME>`
operation.

```

File: _includes/recur.md [hunk 8]
```diff
@@ -37,3 +37,46 @@ Content-Type: application/json
...
+Customers should be able to set `recurrenceToken` to deleted if the consumer 
+deletes their agreements with their customer. 
+Note that the value of `state` must be `Deleted` when deleting the token. 
+No other states are supported.
+
+{:.code-header}
+**Request**
+
+```http
+POST /psp/creditcard/payments HTTP/1.1
```
Comment (asbjornu) [inline _includes/recur.md:50]: This URL does not seem correct to me.

**Review by asbjornu — CHANGES_REQUESTED** [review]
There is a lot of repetition here, so perhaps you can move this to a `delete-token.md` include?

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721&bid=19787468)

File: _includes/delete-token.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    "payeeId": "{{ page.merchant_id }}",
+    "isDeleted": true {% if documentation_section == "card" %}
+    "isPayeeToken": false,
+    "cardBrand": "Visa",
+    "maskedPan": "123456xxxxxx1111",
+    "expiryDate": "MM/YYYY",
+    "tokenType" : "{{ token_field_name }}" {% endif %}
+  }
+}
+```
```
Comment (payex-dev) [inline _includes/delete-token.md:44]: Codacy found an issue: [[no-consecutive-blank-lines] Remove 3 lines after node](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

File: _includes/card-purchase.md [hunk 10]
```diff
@@ -296,6 +296,10 @@ Content-Type: application/json
 | └─➔&nbsp;`href`                   | `string`     | The target URI to perform the operation against.                                                                                                                                                                                                                                                                                                           |
 | └─➔&nbsp;`rel`                    | `string`     | The name of the relation the operation has to the current resource.                                                                                                                                                                                                                                                                                        |
 
+{% include delete-token.md api_resource="creditcard" documentation_section
+="card" token_field_name="paymentToken" %}
```
Comment (asbjornu) [inline _includes/card-purchase.md:300]:
```suggestion
{% include delete-token.md api_resource="creditcard"
documentation_section="card" token_field_name="paymentToken" %}
```

File: _includes/recur.md [hunk 11]
```diff
@@ -60,3 +60,6 @@ Content-Type: application/json
 | {% icon check %} | └─➔&nbsp;`productCategory`     | `string`     | A product category or number sent in from the payee/merchant. This is not validated by Swedbank Pay, but will be passed through the payment process and may be used in the settlement process.                                                                                        |
 | {% icon check %} | └─➔&nbsp;`orderReference`      | `String(50)` | The order reference should reflect the order reference found in the merchant's systems.                                                                                                                                                                                               |
 | {% icon check %} | └─➔&nbsp;`subsite`             | `String(40)` | The subsite field can be used to perform [split settlement][split-settlement] on the payment. The subsites must be resolved with Swedbank Pay [reconciliation][settlement-and-reconciliation] before being used.                                                                      |
+
+{% include delete-token.md api_resource = "creditcard" documentation_section =
+"card" token_field_name ="recurrenceToken" %}
```
Comment (asbjornu) [inline _includes/recur.md:65]:
```suggestion
{% include delete-token.md api_resource="creditcard"
documentation_section="card" token_field_name="recurrenceToken" %}
```

File: payments/invoice/other-features.md [hunk 12]
```diff
@@ -176,6 +176,8 @@ Content-Type: application/json
     }
 }
 ```
+{% include delete-token.md api_resource="invoice" documentation_section =
+"invoice" token_field_name="recurrenceToken" %}
```
Comment (asbjornu) [inline payments/invoice/other-features.md:180]:
```suggestion
{% include delete-token.md api_resource="invoice"
documentation_section="invoice" token_field_name="recurrenceToken" %}
```

**Review by asbjornu — CHANGES_REQUESTED** [review]

**Review by asbjornu — APPROVED** [review]

File: _includes/recur.md [hunk 13]
```diff
@@ -60,3 +60,7 @@ Content-Type: application/json
...
```
Comment (payex-dev) [inline _includes/recur.md:65]: Codacy found an issue: [[no-consecutive-blank-lines] Remove 1 line after node](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721&bid=19787468)

File: _includes/delete-token.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
+{% assign documentation_section = include.documentation_section %}
```
Comment (payex-dev) [inline _includes/delete-token.md:1]: Codacy found an issue: [[final-newline] Missing newline character at end of file](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721&bid=19787468)

File: _includes/recur.md [hunk 15]
```diff
@@ -60,3 +60,7 @@ Content-Type: application/json
 | {% icon check %} | └─➔&nbsp;`productCategory`     | `string`     | A product category or number sent in from the payee/merchant. This is not validated by Swedbank Pay, but will be passed through the payment process and may be used in the settlement process.                                                                                        |
 | {% icon check %} | └─➔&nbsp;`orderReference`      | `String(50)` | The order reference should reflect the order reference found in the merchant's systems.                                                                                                                                                                                               |
 | {% icon check %} | └─➔&nbsp;`subsite`             | `String(40)` | The subsite field can be used to perform [split settlement][split-settlement] on the payment. The subsites must be resolved with Swedbank Pay [reconciliation][settlement-and-reconciliation] before being used.                                                                      |
+
+{% include delete-token.md api_resource="creditcard"
+documentation_section="card" token_field_name="recurrenceToken" %}
```
Comment (payex-dev) [inline _includes/recur.md:65]: Codacy found an issue: [[no-consecutive-blank-lines] Remove 1 line after node](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

**Review by asbjornu — APPROVED** [review]

File: _includes/delete-token.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+{% assign api_resource = include.api_resource %}
+{% assign token_field_name = include.token_field_name %}
+
+Payers should be able to delete payment tokens that are associated to
+them. To delete a `{{ token_field_name }}`, perform the `{{ token_field_name }}`
```
Comment (asbjornu) [inline _includes/delete-token.md:6]: Please find the OPERATION NAME.

```suggestion
them. To delete a `{{ token_field_name }}`, perform the OPERATION NAME
```

**Review by asbjornu — CHANGES_REQUESTED** [review]

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721&bid=19787468)

File: _includes/recur.md (hunk 15 above)
Comment (payex-dev) [inline _includes/recur.md:65]: Codacy found an issue: [[no-consecutive-blank-lines] Remove 1 line after node](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

File: _includes/delete-token.md (hunk 14 above)
Comment (payex-dev) [inline _includes/delete-token.md:1]: Codacy found an issue: [[final-newline] Missing newline character at end of file](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6038721)

Comment (asbjornu) [reply _includes/delete-token.md:1]: Please add a newline to the end of `_includes/delete-token.md`.

File: _includes/delete-token.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
+{% assign documentation_section = include.documentation_section %}
+{% assign api_resource = include.api_resource %}
+{% assign token_field_name = include.token_field_name %}
+
+Payers should be able to delete payment tokens that are associated to
+them. To delete a `{{ token_field_name }}`, view the example below.
```
Comment (asbjornu) [inline _includes/delete-token.md:6]:
```suggestion
them. How to delete a `{{ token_field_name }}` is described in the example below.
```

**Review by asbjornu — CHANGES_REQUESTED** [review]

**Review by asbjornu — APPROVED** [review]

**Pull request merged**